berkeley-function-call-leaderboard/bfcl/model_handler/oss_model/minicpm_fc.py

```diff
@@ -0,0 +1,427 @@
+                + "\n".join(tool_calls).strip()
+                + "\n```\n<|tool_call_end|>\n"
+                + content
+            )
+            # msg["tool_call_string"] = "\n".join(tool_calls).strip()
+            msg["content"] = content
+        else:
+            content = thought_prefix + content
+            msg["content"] = content
+    elif msg["role"] == "user":
+        msg["content"] = user_prefix + "\n" + msg["content"]
+    elif msg["role"] == "system":
+        msg["content"] = msg["content"] + "\n" + system_suffix
+    msg["content"] = msg["content"].strip()
+    return msg
+
+
+def jsonschema_to_code(jsonschema: dict) -> str:
+    input_text = json.dumps(jsonschema)
+    if datamodel_code_generator.get_version() < "0.26.2":
+        from datamodel_code_generator.format import PythonVersion
+
+        data_model_types = get_data_model_types(
+            DataModelType.PydanticBaseModel,
+            target_python_version=PythonVersion.PY_310,
+        )
+    else:
+        from datamodel_code_generator.format import DatetimeClassType, PythonVersion
+
+        data_model_types = get_data_model_types(
+            DataModelType.PydanticBaseModel,
+            target_python_version=PythonVersion.PY_310,
+            target_datetime_class=DatetimeClassType.Datetime,
+        )
+    parser = JsonSchemaParser(
+        source=input_text,
+        data_model_type=data_model_types.data_model,
+        data_model_root_type=data_model_types.root_model,
+        data_model_field_type=data_model_types.field_model,
+        data_type_manager_type=data_model_types.data_type_manager,
+        target_python_version=PythonVersion.PY_311,
+        dump_resolve_reference_action=data_model_types.dump_resolve_reference_action,
+        field_constraints=True,
+    )
+    results = parser.parse()
+    return results
+
+
+def transform_function(function: dict):
+    """turn json format of function into signature"""
+    params, default_params = [], []
+    for prop_name, prop in function["parameters"]["properties"].items():
+        if "default" in prop:
+            default_params.append(f'{prop_name}={repr(prop["default"])}')
+        elif prop_name not in function["parameters"].get("required", []):
+            default_params.append(f"{prop_name}={repr(None)}")
+        else:
+            params.append(prop_name)
+    ps = ", ".join(params + default_params)
+    res = "def {f_name}({ps}):\n".format(f_name=function["name"], ps=ps)
+    f_des = function.get("description", "")
+    content = jsonschema_to_code(function["parameters"])
+    if "class" in content:
+        i = content.index("class")
+        # print(content[:i])
+        content = content[i:]
+    classes, args = content.split("class Model(BaseModel):", 1)
+    lint_msg = f'    """\n    {f_des}\n    Args:\n{args}\n    """\n'
+    res += lint_msg
+    if len(classes) > 0:
+        res = classes + res
+    return res
+
+
+def rename_tool(tool, model_name):
```

Comment:
Good point. I will run a check in the dataset to check for illegal parameter names, and fix the dataset instead of doing conversion in the handler (cause otherwise we have to do it for all models to be fair).

Reply:
implemented in #777 . We will fix all the entries with that issue.